alts/src/main/java/io/grpc/alts/AltsChannelBuilder.java

```diff
@@ -104,7 +105,8 @@ public AltsChannelBuilder enableUntrustedAltsForTesting() {
 
   /** Sets a new handshaker service address for testing. */
   public AltsChannelBuilder setHandshakerAddressForTesting(String handshakerAddress) {
-    HandshakerServiceChannel.setHandshakerAddressForTesting(handshakerAddress);
+    HandshakerServiceChannel.SHARED_HANDSHAKER_CHANNEL.setHandshakerAddressForTesting(
```

Comment:
This is really awkward right now, in that it is a builder-level method that sets a `static`. That's _really_ surprising and seems like a bad idea. I guess that's how it was before, but I just didn't realize it.

Reply:
I don't like this as well. I guess this is one reason we did not use SharedResourceHolder in the first place. Basically, ALTS client needs to connect to the handshaker service (metadata.google.internal:8080). In some testing environment, we override this address to another address, since we may not launch metadata server during testing. We use to use gFlags to control the testing address. But we cannot use gFlags here. 

Let me know if you have better idea. I am not sure I understand your proposal.

Reply:
I provided a better idea: pass in an `ObjectPool` during testing.